sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/DoFnReflectorTest.java

```diff
@@ -357,6 +358,36 @@ public void baz() {}
     });
   }
 
+  private static class PrivateDoFnClass extends DoFnWithContext<String, String> {
+    @ProcessElement
+    public void processThis(ProcessContext c) {}
+  }
+
+  @Test
+  public void testLocalPrivateDoFnClass() throws Exception {
+    underTest(new PrivateDoFnClass());
```

Comment:
I think this just ensures we can create them. To be thorough, we should also call checkInvokeProcessElementWorks.